account_availability.go

```diff
@@ -20,46 +18,24 @@ type AccountAvailability struct {
 	Available []string `json:"available"`
 }
 
-// AccountAvailabilityPagedResponse represents a paginated Account Availability API response
-type AccountAvailabilityPagedResponse struct {
-	*PageOptions
-	Data []AccountAvailability `json:"data"`
-}
-
-// endpoint gets the endpoint URL for AccountAvailability
-func (AccountAvailabilityPagedResponse) endpoint(_ ...any) string {
-	return "/account/availability"
-}
-
-func (resp *AccountAvailabilityPagedResponse) castResult(r *resty.Request, e string) (int, int, error) {
-	res, err := coupleAPIErrors(r.SetResult(AccountAvailabilityPagedResponse{}).Get(e))
-	if err != nil {
-		return 0, 0, err
-	}
-	castedRes := res.Result().(*AccountAvailabilityPagedResponse)
-	resp.Data = append(resp.Data, castedRes.Data...)
-	return castedRes.Pages, castedRes.Results, nil
-}
-
 // ListAccountAvailabilities lists all regions and the resource availabilities to the account.
 func (c *Client) ListAccountAvailabilities(ctx context.Context, opts *ListOptions) ([]AccountAvailability, error) {
-	response := AccountAvailabilityPagedResponse{}
-	err := c.listHelper(ctx, &response, opts)
+	response, err := getPaginatedResults[AccountAvailability](ctx, c, "account/availability", opts)
 	if err != nil {
 		return nil, err
 	}
-	return response.Data, nil
+
+	return response, nil
 }
 
 // GetAccountAvailability gets the resources availability in a region to the customer.
 func (c *Client) GetAccountAvailability(ctx context.Context, regionID string) (*AccountAvailability, error) {
-	req := c.R(ctx).SetResult(&AccountAvailability{})
 	regionID = url.PathEscape(regionID)
 	b := fmt.Sprintf("account/availability/%s", regionID)
```

Comment:
We recently added a new [formatAPIPath](https://github.com/linode/linodego/blob/main/request_helpers.go#L198) helper to automatically handle escaping path segments:

```suggestion
	b := formatAPIPath("account/availability/%s", regionID)
```

Reply:
Thanks! I'll fix this across all of the files in the PR.